### PR TITLE
Abstract out paginated logic in search API views

### PR DESCRIPTION
**Pull request by marcofucci**
@reupen @elcct what do you think about abstracting out the pagination logic?

I've added it just to the OMIS search API View but could be applied to the other ones as well if you want.

This also catches the 500 error about the result window being too big that we are getting from sentry recently.

### CODE REVIEW

**Comment by marcofucci** [comment]
I wanted to use the DRF logic but it's not nicely separated unfortunately.

File: datahub/search/omis/views.py [hunk 1]
```diff
@@ -1,20 +1,65 @@
...
 
 
-class SearchOrderAPIView(APIView):
+class PaginatedAPIMixin:
+    """Mixin for paginated API Views."""
+
+    default_limit = settings.REST_FRAMEWORK['PAGE_SIZE']
+    limit_query_param = 'limit'
+    offset_query_param = 'offset'
+    max_results = 10000
```
Comment (reupen) [inline datahub/search/omis/views.py]: Do we limit the result count returned to the front end to this value then?

Comment (marcofucci) [inline datahub/search/omis/views.py]: Well at the moment it's 10000 so if we change the ES limit we can update this easily
It's not clear to the FE that specifying a value > this raises 500 I think and it's also not nice to receive those errors in general.

Comment (marcofucci) [inline datahub/search/omis/views.py]: oh sorry, I understand the problem now. I think it's a good idea to limit the number of pages to this overall value. Anything after this value is probably not very useful anyway.

Comment (reupen) [inline datahub/search/omis/views.py]: Hmm, might not be ideal on the collection pages when no or few filters are applied though.

Comment (marcofucci) [inline datahub/search/omis/views.py]: @reupen OK should I take out the `max_results` and the validation check and we keep it for later?

Comment (reupen) [inline datahub/search/omis/views.py]: No strong opinion. As long as we remember to do something about it...

**Pull request merged**